builtin/logical/ssh/path_sign.go

```diff
@@ -228,6 +228,29 @@ func (b *backend) calculateValidPrincipals(data *framework.FieldData, defaultPri
 		if principalsAllowedByRole == "*" {
 			return parsedPrincipals, nil
 		}
+		// Role was configured to allow current user as principal.
+		if principalsAllowedByRole == "?" {
+			// Use the authenticated users entity alias as username.
+			var username string
+			if req.EntityID != "" {
+				entity, err := b.System().EntityInfo(req.EntityID)
+				if err != nil {
+					return nil, err
+				}
+				if len(entity.Aliases) > 0 {
+					// Take the first alias available for the username.
```

Comment:
Is it worth considering setting the entity name to the entity alias's name when a brand new entity is created for an alias?

i.e. in this code: https://github.com/hashicorp/vault/blob/c27a1c1840a4f73674c5a511b7a2ed5a44cc73bb/vault/identity_store.go#L530-L531, if entity.Name were set to the entity alias name after the entity struct was initialized, then the code here (and other similar use cases) would not need to guess/use their own heuristic to find a useful principal name from the entity's aliases.

Reply:
Using identity template.